diff --git a/include/factory.hpp b/include/factory.hpp
index ca707a3..2e9f842 100644
--- a/include/factory.hpp
+++ b/include/factory.hpp
@@ -1,4 +1,9 @@
 #pragma once
+#define HAVE_WLR
+#define HAVE_HYPRLAND
+#define HAVE_RIVER
+#define HAVE_SWAY
+#define USE_EXPERIMENTAL
 
 #include <json/json.h>
 #ifdef HAVE_LIBDATE
diff --git a/src/bar.cpp b/src/bar.cpp
index f46b7d0..dcd9dec 100644
--- a/src/bar.cpp
+++ b/src/bar.cpp
@@ -595,7 +595,7 @@ waybar::Bar::Bar(struct waybar_output* w_output, const Json::Value& w_config)
 
   window.signal_map_event().connect_notify(sigc::mem_fun(*this, &Bar::onMap));
 
-#if HAVE_SWAY
+#ifdef HAVE_SWAY
   if (auto ipc = config["ipc"]; ipc.isBool() && ipc.asBool()) {
     bar_id = Client::inst()->bar_id;
     if (auto id = config["id"]; id.isString()) {
